frontend/src/utils/date-utils.ts

```diff
@@ -10,7 +10,13 @@ export const datesAreEqualWithoutTime = (
   date1: number | Date,
   date2: number | Date,
 ): boolean => {
-  return new Date(date1).toDateString() === new Date(date2).toDateString();
```

Comment:
This is not visible at startup, but when clicking around the app a few times with a layer open (change date with the bottom selector, open a popup, etc...), this function consumes a lot of CPU time (~1s after 4-5 clicks). The new code is about 27% faster according to benchmark.

Reply:
Maybe worth extracting into a compareDate function? It might already exist in some places actually

Reply:
Changed it again, now it's about twice as fast. I haven't found any `compareDates` function, but it's quite likely that there is one somewhere indeed :)